### PR TITLE
Provide accessor to the span implementation

### PR DESCRIPTION
**Pull request by dashpole**
This is needed if the SDK needs to cast to its underlying type.

Part of https://github.com/open-telemetry/opentelemetry-go/issues/93

@nilebox

### CODE REVIEW

File: trace/trace_api.go [hunk 1]
```diff
@@ -148,6 +148,11 @@ type Span struct {
 	internal SpanInterface
 }
 
+// Internal returns the underlying implementation of the Span
+func (s *Span) Internal() SpanInterface {
```
Comment (nilebox) [inline trace/trace_api.go:152]: Can you please explain why is this necessary?

**Comment by dashpole** [comment]
This is needed when attempting to add an OpenCensus span to the OpenTelemetry context using `otel.ContextWithSpan(context.Context, otel.Span)`  In order to do that, we need to be able to get the underlying Span implementation.  The full implemnetation would look something like:


```golang
// This is the OpenCensus FromContext function, implemented using opentelemetry's SpanFromContext
func (o *otelTracer) FromContext(ctx context.Context) *octrace.Span {
	otSpan := otel.SpanFromContext(ctx)
	return // convert the opentelemetry span into the OC span implementation
}

// This is the OpenCensus NewContext function, implemented using opentelemetry's ContextWithSpan
func (o *otelTracer) NewContext(parent context.Context, s *octrace.Span) context.Context {
        // Below, I need to call s.Internal() in order to have an interface I can cast to my internal type.
        // For example, my internal type could wrap an opentelemetry Span, which I could use in ContextWithSpan
	if myspan, ok := s.Internal().(*myspanimplementation); ok {
		return otel.ContextWithSpan(parent, // get an opentelemetry Span from myspan)
	}
	// The user must have created the octrace Span using a different tracer, and we don't know how to store it.
	return parent
}
```

File: trace/trace_api.go (hunk 1 above)
Comment (dashpole) [reply trace/trace_api.go:152]: This is needed when attempting to add an OpenCensus span to the OpenTelemetry context using `otel.ContextWithSpan(context.Context, otel.Span)`  In order to do that, we need to be able to get the underlying Span implementation.  The full implemnetation would look something like:


```golang
// This is the OpenCensus FromContext function, implemented using opentelemetry's SpanFromContext
func (o *otelTracer) FromContext(ctx context.Context) *octrace.Span {
	otSpan := otel.SpanFromContext(ctx)
	return // convert the opentelemetry span into the OC span implementation
}

// This is the OpenCensus NewContext function, implemented using opentelemetry's ContextWithSpan
func (o *otelTracer) NewContext(parent context.Context, s *octrace.Span) context.Context {
        // Below, I need to call s.Internal() in order to have an interface I can cast to my internal type.
        // For example, my internal type could wrap an opentelemetry Span, which I could use in ContextWithSpan
	if myspan, ok := s.Internal().(*myspanimplementation); ok {
		return otel.ContextWithSpan(parent, // get an opentelemetry Span from myspan)
	}
	// The user must have created the octrace Span using a different tracer, and we don't know how to store it.
	return parent
}
```

Comment (nilebox) [reply trace/trace_api.go:152]: Oh I see that makes sense.

**Review by nilebox — APPROVED** [review]
